### ISSUE TITLE
(Creature/Vendors): Jesse Masters cursor icon issue.

### ISSUE THREAD

**Issue by heyitsbench** [azerothcore/azerothcore-wotlk#15812]
### Current Behaviour

Currently if you haven't completed the quests [Replenishing the Storehouse](https://wowgaming.altervista.org/aowow/?quest=12212) and [Take Their Rear!](https://wowgaming.altervista.org/aowow/?quest=12216), hovering your cursor over [Jesse Masters](https://wowgaming.altervista.org/aowow/?npc=29244) will show the vendor bag icon despite him not selling anything to you.

### Expected Blizzlike Behaviour

In the above situation, the player should see the chat bubble cursor icon rather than the vendor bag.

### Source

Wrath Classic

### Steps to reproduce the problem

1. `.tele amberpinelodge`
2. Interact with Masters.
3. Observe your cursor as you do so.

### Extra Notes

This is almost definitely not just an issue with Jesse Masters, but an overall vendor issue. I only explicitly mention this creature because he is an easy example, as I observed the behavior during the exchange with [this issue](https://github.com/azerothcore/azerothcore-wotlk/issues/15771) and [this PR](https://github.com/azerothcore/azerothcore-wotlk/pull/15777).

### AC rev. hash/commit

686055c54757af866d8851a9bf6665ccc547f1f2

### Operating system

Windows 10 x64

### Custom changes or Modules

[mod-ptr-template](https://github.com/heyitsbench/mod-ptr-template)

**Comment by heyitsbench** [comment]
To be clear, after finishing the first quest (Replenishing the Storehouse) and unlocking the vendor menu in the first place, the cursor icon should then become the vendor bag.

**Comment by Kitzunu** [comment]
This is most likely a generic issue, and it is showing the cursor due to the npcflags he's got

**Comment by heyitsbench** [comment]
Yep, I can confirm Koren in Karazhan (vendor gated by Violet Eye reputation) is affected by this as well.

**Comment by heyitsbench** [comment]
Despite this seeming to be a global behavior, I can confirm the Exodar Peacekeeper spawned by the fire brigade practice area has the typical guard (ones that give directions) cursor despite not giving directions.
